Sming/Components/Network/Arch/Esp32/Platform/StationImpl.cpp

```diff
@@ -418,12 +418,12 @@ void StationImpl::staticScanCompleted(wifi_event_sta_scan_done_t* event, uint8_t
 		if(station.scanCompletedCallback) {
 			uint16_t number = event->number;
 			wifi_ap_record_t ap_info[number];
-			uint16_t ap_count{0};
+
 			memset(ap_info, 0, sizeof(ap_info));
 			ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(&number, ap_info));
-			ESP_ERROR_CHECK(esp_wifi_scan_get_ap_num(&ap_count));
+
 			// TODO: Handle hidden APs
-			for(unsigned i = 0; (i < event->number) && (i < ap_count); i++) {
+			for(unsigned i = 0; (i < event->number); i++) {
```

Comment:
I think this loop should use `number` instead of `event->number` as it's the more recently aquired value. Presumably these values should always be the same, so could include an `assert(number == event->number)` but I think that's overkill.

Reply:
ack, especially as it's unclear if `esp_wifi_scan_get_ap_records` changes `number` and what it would mean if those diverge